### ISSUE TITLE
jax.grad returns zero for i0'(1e-300) instead of finite float64 derivative

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#39797]
### Description

`jax.grad` appears to return an incorrect zero gradient for `jax.numpy.i0` at a very small positive `float64` input.

The derivative of the modified Bessel function `i0(x)` is `i1(x)`. Around zero, `i0'(x)` is approximately `x / 2`. For `x = 1e-300`, the expected derivative is approximately `5e-301`, which is finite and representable in `float64`.

However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = 1e-300
expected = 5e-301

out = float(jax.grad(lambda z: jnp.i0(z))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: 5e-301
```

### Expected result

```
5e-301
```

The expected value is finite and representable in `float64`. Since `d/dx i0(x) = i1(x)`, and `i1(x) ~ x / 2` near zero, the gradient at `x = 1e-300` should be approximately `5e-301`, not zero.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
#38651
